### PR TITLE
Update installation doc

### PR DESCRIPTION
**Pull request by AmishgaAlphonius**
### Description

This PR follows #1181 and updates documentation related to the installation of Lethe. More specifically, it corrects the version of _p4est_ to be installed to 2.3.6.

### Miscellaneous (will be removed when merged)
- The `STABLE_BUILD` variable in `candi.cfg` was depricated. To get the developement release, simply specify the version `master`.
- The updated version of Windows installation has not been tested yet.
- The updated version of ARM installation has not been tested yet.

### Checklist (will be removed when merged)
See [this page](https://chaos-polymtl.github.io/lethe/documentation/contributing.html#pull-requests) for more information about the pull request process.

Code related list:
- [x] All in-code documentation related to this PR is up to date (Doxygen format)
- [x] Lethe documentation is up to date
- [x] The branch is rebased onto master
- [x] Code is indented with indent-all and .prm files (examples and tests) with prm-indent

Pull request related list:
- [x] Labels are applied
- [x] There are at least 2 reviewers (or 1 if small feature) excluding the responsible for the merge
- [x] If this PR closes an issue or is related to a project, it is linked in the "Projects" or "Development" section
- [x] If any future works is planed, an issue is opened
- [x] The PR description is cleaned and ready for merge

### CODE REVIEW

File: doc/source/installation/windows_wsl.rst [hunk 1]
```diff
@@ -234,21 +234,25 @@ Do not forget the ``.`` at the end of the command, which means "here".
...
+  |     12 | ``CHECKSUM="${CHECKSUM} 0ea6e4806b6...``       | ``#CHECKSUM="${CHECKSUM} 0ea6e4806b6...``                                     |
+  +--------+------------------------------------------------+-------------------------------------------------------------------------------+
+  |     13 |                                                | .. code-block:: text                                                          |
+  |        |                                                |   :class: copy-button                                                         |
+  |        |                                                |                                                                               |
+  |        |                                                |   VERSION=2.3.6                                                               |
+  |        |                                                |   CHECKSUM=4b35d9cc374e3b05cd29c552070940124f04af8f8e5e01ff046e39833de5e153   |
+  +--------+------------------------------------------------+-------------------------------------------------------------------------------+
```
Comment (oguevremont) [inline doc/source/installation/windows_wsl.rst:255]: I like this format for adding new lines. I suggest we write it the same way for regular installation and Apple ARM

File: doc/source/installation/apple_arm.rst [hunk 2]
```diff
@@ -24,7 +24,29 @@ To install the dependencies (mpi, p4est, trilinos and METIS) all together using
 
   sudo apt-get install libmuparser-dev
 
-Clone the candi git repository in a folder of your choice  (e.g. ``$HOME/sofware/``). You can edit the ``candi.cfg`` file if you want to force the installation of the deal.II master version instead of the current stable version by setting the ``STABLE_BUILD=false``. Under Apple ARM, we only recommend the installation of the required libraries, namely parmetis, trilinos and p4est.
+Clone the candi git repository in a folder of your choice  (e.g. ``$HOME/software/``). You can edit the ``candi.cfg`` file if you want to force the installation of the deal.II master version instead of the current stable version by setting ``DEAL_II_VERSION=master`` on line 97. Under Apple ARM, we only recommend the installation of the required libraries, namely parmetis, trilinos and p4est.
+
+To ensure that the Lethe test suite works, deal.II must be configured with p4est version 2.3.6. In the subfolder ``deal.II-toolchain/packages/``, open the ``p4est.package`` file with a text editor and change the following lines:
+
+- Comment lines 9 to 12:
+
```
Comment (oguevremont) [inline doc/source/installation/apple_arm.rst:32]:
```suggestion

```

**Review by oguevremont — APPROVED** [review]

File: doc/source/installation/apple_arm.rst [hunk 3]
```diff
@@ -24,7 +24,29 @@ To install the dependencies (mpi, p4est, trilinos and METIS) all together using
...
+  +--------+------------------------------------------------+-------------------------------------------------------------------------------+
+  | line # | initial line                                   | changed line                                                                  |
+  +========+================================================+===============================================================================+
+  |     9  | ``VERSION=2.3.2``                              | ``#VERSION=2.3.2``                                                            |
```
Comment (acdaigneault) [inline doc/source/installation/apple_arm.rst:36]: Maybe we should just write to remove all that section and to add ``VERSION=2.3.6`` ?

Comment (acdaigneault) [inline doc/source/installation/apple_arm.rst]: Maybe we should just write to remove all that section and to add ``VERSION=2.3.6`` ?

File: doc/source/installation/windows_wsl.rst (hunk 1 above)
Comment (AmishgaAlphonius) [reply doc/source/installation/windows_wsl.rst:255]: I was hesistating between both format, so I thought leaving them both here, the reviewers will tell me their opinion.

File: doc/source/installation/apple_arm.rst (hunk 3 above)
Comment (acdaigneault) [reply doc/source/installation/apple_arm.rst]: Also, the candi script changes over time, what is here may change

**Review by acdaigneault — COMMENTED** [review]
I just want to add that the Trilinos version in the current Candi script had change to 14.0, which also require cmake 3.26 if you have an older version.
Do you think we could fork candi and modify everything according to our parameters?

**Comment by AmishgaAlphonius** [comment]
> I just want to add that the Trilinos version in the current Candi script had change to 14.0, which also require cmake 3.26 if you have an older version. Do you think we could fork candi and modify everything according to our parameters?

I think that's a great idea to make a fork, it would simplify the steps for users that are installing and therefore make it more easy (accessible) for them!
If @blaisb is fine with it, I could do so.
As for the Trilinos version, I was hesistating over it also, but as it is machine-dependant prefered not forcing a version, but I could change it for 13 on the fork (or doc), and leave a comment about it.

**Review by OresteMarquis — APPROVED** [review]
Nothing to add, changes are clear! I reinstalled deal.ii and Lethe on my windows computer at home using wsl and candi and everything worked!

File: doc/source/installation/apple_arm.rst (hunk 3 above)
Comment (OresteMarquis) [reply doc/source/installation/apple_arm.rst]: When reinstalling I indeed needed to update my cmake. I think if we want more control over the installation guide using candi a fork could be a good idea!

**Comment by blaisb** [comment]
I agree with making a fork for candi, it's a good idea. I'm also a maintainer of CANDI so I could also make changes directly into the repository if we want

File: doc/source/installation/windows_wsl.rst [hunk 4]
```diff
@@ -297,7 +301,7 @@ Where ``$numprocs`` corresponds to the number of processors used for the compila
 Installing Lethe (Step #2)
 -------------------------------------
 
-1. |linux_shell| Set-up the folder structure. Create the ``Software`` folder (if you are doing the candi installation, this folder should alredy exist from Step #1).
+1. |linux_shell| Set-up the folder structure. Create the ``Software`` folder (if you are doing the candi installation, this folder should already exist from Step #1).
```
Comment (blaisb) [inline doc/source/installation/windows_wsl.rst:304]: The only thing I don't like is that Software here is with a capital letter. Haha. Otherwise, chill

**Review by blaisb — APPROVED** [review]
Great work :)! I think this is ready for a merge.

**Pull request merged**